src/app/domain/geolocation/hooks/tracking.ts

```diff
@@ -14,10 +22,25 @@ export const useGeolocationTracking = (): void => {
     const initializeTracking = async (): Promise<void> => {
       if (!client) return
 
-      const shouldStartTracking = (await getShouldStartTracking()) as boolean
+      const trackingEnabled = await isGeolocationTrackingEnabled()
+
+      if (trackingEnabled) {
```

Comment:
this part might be too nested, it's a bit hard to grok at first sight what's going on, maybe it can be refactor with named functions etc.

Also is there error handling above this hook? We can see a good number of async operations which could be points of failure. But maybe it is handled inside the async functions themselves

Reply:
I push two new commits with named function it is very clean now !